protocol/0053-PERP-product_builtin_perpetual_future.md

```diff
@@ -28,23 +53,71 @@ cash_settled_perpetual_future.value(quote) {
-	setMarkPrice(event.data)
+If the periodic settlement data received satisfies all the filters that have been specified for it then that data point (`y`) along with the current `mark_price` (`x`) for the market and the current `vegaprotocol.builtin.timestamp` (`t`) gets stored as the funding payment data point.
+
+### 4.2. Mark to market settlement
+
+Every time a [mark to market settlement](./0003-MTMK-mark_to_market_settlement.md) is carried out the value of the last periodic settlement data point received along with the price used for MTM settlement and the current `vegaprotocol.builtin.timestamp` gets stored as the funding payment data point. If no periodic settlement data has been received yet then the funding payment data point should not be created.
+
+### 4.3. Periodic settlement
+
+When the `settlement_schedule` event is received the latest funding payment data point gets repeated with the timestamp set to the current value of `vegaprotocol.builtin.timestamp`.
+
+The next step is to calculate the periodic settlement funding payment. If there are no periodic settlement data points then the periodic settlement is skipped. Otherwise, consider all the periodic settlement data points and calculate the time-weighted average price difference as:
+
+```go
+sd := 0
+st := 0
+for i := 0; i < len(data_points) - 1; i++ {
+    t := data_points[i+1].t-data_points[i].t
+    d := data_points[i].x - data_points[i].y     // recall that x stands for mark price and y for the external price source input
+    sd += d * t
+    st += t
 }
+funding_payment = sd / st
 ```
 
+All the funding payment data points except for the last one (it should get carried over as the first data point for the next period) can then be deleted.
+
+Last step is to calculate each party's cash flows as $-\text{open volume} * \text{funding payment}$ where cashflows are first collected from parties that are making the payment (negative value of the cashflow, i.e. longs when the funding payment is positive) and distributed to those receiving it. Any shortfall should be made-up from the insurance pool and if that's not possible loss socialisation should be applied (exactly as per mark-to-market settlement methodology).
+
+### 4.3.1. Periodic settlement during [auction](0026-AUCT-auctions.md)
+
+Periodic settlement is not allowed during the opening auction and it's extensions.
+If periodic settlement data happens whilst market is in auction of any other type then periodic settlement should be carried out as per above methodology and the market should remain in auction until it's allowed to move back into market's default trading mode.
+
+### 5. Margin considerations
+
+To assure adequate solvency we need to include the estimate of the upcoming funding payment in maintenance margin estimate for the party. Let $t_{k-1}$ be the time of the last funding payment. Let $t$ be current time ($t < t_k$).
+Calculate $G_t$ as the [funding payment](#44-periodic-settlement) between $t_k$ and $t$.
```

Comment:
Is this suppose to be `between $t_{k-1}$ and $t$.`?

Reply:
```suggestion
Calculate $G_t$ as the [funding payment](#44-periodic-settlement) between $t_{k-1}$ and $t$.
```